plugins/flytekit-omegaconf/flytekitplugins/omegaconf/type_information.py

```diff
@@ -0,0 +1,124 @@
+import dataclasses
+import logging
+import typing
+from collections import ChainMap
+
+from dataclasses_json import DataClassJsonMixin
+
+from omegaconf import DictConfig, ListConfig, OmegaConf
+
+logger = logging.getLogger("flytekit")
+
+NoneType = type(None)
+
+
+def substitute_types(t: typing.Type) -> typing.Type:
+    """
+    Provides a substitute type hint to use when selecting transformers for serialisation.
+
+    Args:
+        t: Original type
+
+    Returns:
+        A corrected typehint
+    """
+
+    if hasattr(t, "__origin__"):
+        # Only encode generic type and let appropriate transformer handle the rest
+        if t.__origin__ in [dict, typing.Dict]:
+            t = DictConfig
+        elif t.__origin__ in [list, typing.List]:
+            t = ListConfig
+        else:
+            return t.__origin__
+    return t
+
+
+def all_annotations(cls: typing.Type) -> ChainMap:
+    """
+    Returns a dictionary-like ChainMap that includes annotations for all
+    attributes defined in cls or inherited from superclasses.
+    """
+    return ChainMap(*(c.__annotations__ for c in cls.__mro__ if "__annotations__" in c.__dict__))
+
+
+def extract_node_type(
+    python_val: typing.Union[DictConfig, ListConfig], key: typing.Union[str, int]
+) -> typing.Tuple[type, str]:
+    """
+    Provides typing information about DictConfig nodes
+
+    Args:
+        python_val: A DictConfig
+        key: Key of the node to analyze
+        mode: The mode in which to serialize type information
+
+    Returns:
+        Type - The extracted type
+        str - String representation for (de-)serialisation
+
```

Comment:
Reformat this docstring to use the [reST format](http://daouzli.com/blog/docstring.html#id26) (e.g. [[1]](https://github.com/flyteorg/flytekit/blob/aa0b7c70450a5ff65956e4964f48f33649a25233/flytekit/configuration/file.py#L298-L304) or [[2]](https://github.com/flyteorg/flytekit/blob/aa0b7c70450a5ff65956e4964f48f33649a25233/flytekit/core/array_node_map_task.py#L44-L51)).